pkg/helpers/pki_test.go

```diff
@@ -235,3 +235,15 @@ func TestCreatePkiKeyCertPair(t *testing.T) {
 		t.Errorf("unexpected error thrown while executing CreatePkiKeyCertPair : %s", err.Error())
 	}
 }
+
+func TestSetSetPkiKeySize(t *testing.T) {
+	if GetPkiKeySize() != DefaultPkiKeySize {
+		t.Errorf("the default size is not correct.")
+	}
+
+	// set to the default size so it will have no impact on other testing.
+	SetPkiKeySize(DefaultPkiKeySize)
```

Comment:
this test isn't really testing functionality since the expected value doesn't change before/after the set... Could we maybe set it to something different then set it back to 0?

Reply:
yes, I ever thought about it, but I think it may cause some problem if the testing is running in multi-thread mode.

Reply:
in any case, we should undo all changes after the test runs and set back the override size to `0` so the default is used.

If we aren't able to set the value without impacting other tests running in parallel maybe it's a sign that a public var isn't the best way to expose this... Maybe it should be a parameter in SetPropertiesDefault instead?

Reply:
yes, add it into the SetPropertiesDefault method is better.
but that would be one API change.
would that be too big for this kind of purpose?

if you agree, I can move to that part :)

Reply:
change the API means a lot of the unit tests code change.
and even for the "projects" which reference aks-engine :)

Reply:
Yes, on second thought maybe changing the SetPropertiesDefault function signature isn't the best idea since it's used in so many places. @devigned do you have any thoughts on this?